experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,296 @@
+exceed 64000 bytes
+
+##### flags, two bytes
+
+###### flags[0]
+
+```
+- 7 (MSB): unused
+- 6: unused
+- 5: unused
+- 4: unused
+- 3: unused
+- 2: unused
+- 1: end flag. MUST be set in the last frame of a message
+- 0 (LSB): start flag. MUST be set in the first frame of a message
+```
+
+###### flags[1]
+
+```
+- 7 (MSB): unused
+- 6: unused
+- 5: unused
+- 4: unused
+- 3: unused
+- 2: unused
+- 1: unused
+- 0 (LSB): unused
+```
+
+##### body, variable length byte array
+
+Byte chunk of an encoded message. In order to not exceed the maximum frame
+length, this byte array MUST NOT exceed 63,936 bytes
+
+## Examples for Illustration
+
+### Frame Flags
+
+For messages sent as more than two frames, the first frame would have the start
+flag set, all other flags would be unset. The last frame would have the end flag
+set, all other flags would be unset. Any frames in between would have no flags
+set.
+
+For small messages sent as a single frame, the start flag and the end flag would
+be set, all other flags would be unset.
+
+### Layers Diagram
+
+The definition section defines three layers:
+
+- Method Layers: Defines the available methods, parameter and return value
+  encoding
+- Message Layer: Defines Request Messages and Response Messages
+- Frame Layer: Defines how Messages are split into frames
+
+The diagram below shows a practial example of how a method invocation would look
+like after being encoded as two frames.
+
+#### Frame 0
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         frame padding                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|          frame length         |          frame flags          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 frame body [ message length ]                 |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|              frame body [ message invocation_id ]             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                frame body [ message method_id ]               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 frame body [ message padding ]                |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++   frame body [ message body [ method encoded_parameters ] ]   +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ....
+```
```

Comment:
Generated from the schema `frame padding:32,frame length:16,frame flags:16,frame body [ message length ]:32,frame body [ message invocation_id ]:32,frame body [ message method_id ]:32,frame body [ message padding ]:32,frame body [ message body [ method encoded_parameters ] ]:128`